### PR TITLE
[CORE] Add nativeFilters info for simpleString of scan

### PR DESCRIPTION
**Pull request by zml1206**
## What changes were proposed in this pull request?
SQL: `select l_partkey from lineitem where l_partkey + 1 > 5 and rand() < 0.5`
Before pr:
V1:
image
V2:
image

After pr:
V1:
image
V2:
image

### CODE REVIEW

**Comment by zhztheplayer** [comment]
@zml1206 Would you like to add an example to demonstrate on the difference made by the change? Thanks!

File: gluten-substrait/src/main/scala/org/apache/gluten/execution/BatchScanExecTransformer.scala [hunk 1]
```diff
@@ -169,4 +170,13 @@ abstract class BatchScanExecTransformerBase(
     case "ClickHouseScan" => ReadFileFormat.MergeTreeReadFormat
     case _ => ReadFileFormat.UnknownFormat
   }
+
+  override def simpleString(maxFields: Int): String = {
+    val truncatedOutputString = truncatedString(output, "[", ", ", "]", maxFields)
+    val runtimeFiltersString = s"RuntimeFilters: ${runtimeFilters.mkString("[", ",", "]")}"
+    val nativeFiltersString = s"nativeFilters: ${filterExprs().mkString("[", ",", "]")}"
```
Comment (zhztheplayer) [inline gluten-substrait/src/main/scala/org/apache/gluten/execution/BatchScanExecTransformer.scala:177]: Nit: Do we prefer `NativeFilters` than `nativeFilters`?

File: gluten-substrait/src/main/scala/org/apache/gluten/execution/FileSourceScanExecTransformer.scala [hunk 2]
```diff
@@ -190,6 +193,18 @@ abstract class FileSourceScanExecTransformerBase(
...
       case _ => ReadFileFormat.UnknownFormat
     }
+
+  override def simpleString(maxFields: Int): String = {
+    val metadataEntries = metadata.toSeq.sorted.map {
+      case (key, value) =>
+        key + ": " + StringUtils.abbreviate(redact(value), maxMetadataValueLength)
+    }
+    val metadataStr = truncatedString(metadataEntries, " ", ", ", "", maxFields)
+    val nativeFiltersString = s"nativeFilters: ${filterExprs().mkString("[", ",", "]")}"
```
Comment (zhztheplayer) [inline gluten-substrait/src/main/scala/org/apache/gluten/execution/FileSourceScanExecTransformer.scala:203]: ditto

**Review by zhztheplayer — APPROVED** [review]
Just one minor. Thank you.

File: gluten-substrait/src/main/scala/org/apache/gluten/execution/BatchScanExecTransformer.scala (hunk 1 above)
Comment (zml1206) [reply gluten-substrait/src/main/scala/org/apache/gluten/execution/BatchScanExecTransformer.scala]: Good catch, thank you.

**Pull request merged**